### PR TITLE
Support different kernel repo (BugFix)

### PR DESCRIPTION
**Pull request by baconYao**
## Description

Propose an environment variable called **KERNEL_REPO** to address issue. The default Kernel repo is still from**Main**, but provide a flexibility way for alternative.

## Resolved issues
Solve issue: #977 

## Documentation

Ref Jira Card: https://warthogs.atlassian.net/browse/OEMQA-3922

## Tests

- G1200-evk P1V2 (CID: 202307-31859)
  - Pass Result with defining the KERNEL_REPO=”universe” in config_vars
    - https://certification.canonical.com/hardware/202307-31859/submission/354476/test-results/pass/
  - Failed Result since without defining the KERNEL_REPO=”universe” in config_vars
    - https://certification.canonical.com/hardware/202307-31859/submission/354475/test-results/fail/
- BMX13-PV-SKU14 (CID: 202312-33305)
  - Pass Result is expected for OEM-PC (main Ubuntu repository)
    - https://certification.canonical.com/hardware/202312-33305/submission/354484/test-results/pass/

### CODE REVIEW

File: providers/base/bin/check_prerelease.py [hunk 1]
```diff
@@ -27,72 +27,135 @@
...
+        # "apt-cache showpkg" returns an empty string with exit code 0 if
+        # called on a non-existent package.
+        if aptinfo == '':
+            raise CalledProcessError(returncode=None, cmd=None)
+        return aptinfo
+    except CalledProcessError:
+        # "apt-cache show" returns an error status if called on a
+        # non-existent package.
+        logging.error(
+            "* Kernel does not match any installed package!")
```
Comment (pieqq) [inline providers/base/bin/check_prerelease.py:70]:
```suggestion
        if aptinfo == "":
            raise CalledProcessError(returncode=1, cmd=command)
        return aptinfo
    except CalledProcessError as e:
        # "apt-cache show" returns an error status if called on a
        # non-existent package.
        logging.error(e)
        logging.error(
            "* Kernel does not match any installed package!")
```

I feel like it's better to raise `CalledProcessError` with a proper returncode and command; and we can actually use this to show more information in the output (with `logging.error(e)`), which could help if anything goes wrong during testing.

**Review by pieqq — APPROVED** [review]
Nice work, Patrick! I've poked at the unit tests and did a few tests on my machine and it works as expected here.

I made one suggestion, but this is not a requirement. We can land this as is. Let me know what you think!

File: providers/base/bin/check_prerelease.py (hunk 1 above)
Comment (baconYao) [reply providers/base/bin/check_prerelease.py:70]: Nice suggestion, I like it.

**Review by pieqq — APPROVED** [review]
Awesome! Thanks!

**Pull request merged**